docs/source/pages/endpoints.rst

```diff
@@ -0,0 +1,123 @@
+=========
+Endpoints
+=========
+
+Sanic JWT sets itseld up to run as a `Sanic Blueprint <http://sanic.readthedocs.io/en/latest/sanic/blueprints.html>`_ at the ``/auth`` path. ::
+
+    http://localhost:8000/auth
+
+This is can be changed via the ``SANIC_JWT_URL_PREFIX`` setting. :doc:`See settings for more <settings>`. ::
+
+    app.config.SANIC_JWT_URL_PREFIX = '/api/authentication'
+
+All Sanic JWT endpoints will now be available at: ::
+
+    http://localhost:8000/api/authentication
+
+------------
+
+++++++++++++
+Authenticate
+++++++++++++
+
+| **Path**: ``/auth``
+| **Acceptable Methods**: ``POST``
+| **Purpose**: Generates an access token if the ``authenticate`` :doc:`method <initialization>` evaluates to ``True``.
+| **Example**:
+|
+
+Request ::
+
+    curl -X POST -H "Content-Type: application/json" -d '{"username": "<USERNAME>", "password": "<PASSWORD>"}' http://localhost:8000/auth
+
+Response ::
+
+    {
+        "access_token": "<JWT>"
+    }
+
+------------
+
+++++++++++++
+Verification
+++++++++++++
+
+| **Path**: ``/auth/verify``
+| **Acceptable Methods**: ``GET``
+| **Purpose**: Check whether or not a given access token is valid.
+| **Example**:
+|
+
+Request ::
+
+    curl -X GET -H "Authorization: Bearer <JWT>" http://localhost:8000/auth/verify
+
+Response ::
+
+    200 Response
+    {
+        "valid": true
+    }
+
+    ## or
+
+    400 Response
+    {
+        "valid": false,
+        "reason": "Signature has expired"
+    }
+
+------------
+
+++++++++++++++++++++
+Current User Details
+++++++++++++++++++++
+
+| **Path**: ``/auth/me``
+| **Acceptable Methods**: ``GET``
+| **Purpose**: Retrieve information about the currently authenticated user.
+| **Example**:
+|
+
+Request ::
+
+    curl -X GET -H "Authorization: Bearer <JWT>" http://localhost:8000/auth/me
+
+Response ::
+
+    200 Response
+    {
+        "user_id": 123456
+    }
+
+
+.. note::
+
+    Because this package does not know about you user management layer, you need to have a user object that either is a dict or a object instance with a ``to_dict()`` or ``__dict__`` method. The output of these methods will be used to generate the ``/me`` response.
```

Comment:
I think we should mention that `to_dict` will be preferable over `__dict__` since there are a lot of third party classes that implements this (and sometimes depends on it). One example would be using an ORM and returning the user object, where `__dict__` is mostly likely to be available - and the result would not be, well, not "advisable" so to speak. Quick example using SQLAlchemy:

```python
class User(Base):
    __tablename__ = 'users'

    id = Column(Integer, primary_key=True)
    name = Column(String)
    fullname = Column(String)
    password = Column(String)


some_user = session.query(User).first()
print(hasattr(some_user, '__dict__'))
# >> True
print(dict(some_user))
# >> {'_sa_instance_state': <sqlalchemy.orm.state.InstanceState object at 0x7fb9de99f780>}
```

Reply:
Good point.  Maybe the name of this too should be configurable for precisely that reason.

Reply:
I know that `ujson` requires (?) a method named `__json__` [to be implemented](https://github.com/esnme/ultrajson/blob/master/python/objToJSON.c#L634) if you want to serialize your objects through their interface, but it is hardcoded in C. We can be more flexible (by configuring the method name); stick to what's already running `isinstance(x, dict)` and / or `to_dict` or even use something completely different, like `to_jwt` ... The first being my personal choice.

Reply:
I was searching for some `dict` stuff and found [this thread](https://stackoverflow.com/questions/25440694/whats-the-purpose-of-dictproxy) on stackoverflow. It gives a more in depth (tldr included) knowledge about how `__dict__` works ... IMHO, I think we should drop the `__dict__` idea for good.